lib/ansible/executor/module_common.py

```diff
@@ -65,6 +66,8 @@ def _find_snippet_imports(module_data, module_path, strip_comments):
     module_style = 'old'
     if REPLACER in module_data:
         module_style = 'new'
+    elif REPLACER_WINDOWS in module_data:
+        module_style = 'new'
```

Comment:
How about the change I added [below](https://github.com/ansible/ansible/pull/11719/files#diff-9584222c646a5fc71bf7537da48741e6R141) that adds the ability for a Windows module to indicate if it supports check mode?  With this change, check mode would not be supported by any Windows modules initially (except `setup.ps1` because it doesn't call `Parse-Args`), but it would be easy to [add that flag to the modules that don't modify the remote system](https://github.com/cchurch/ansible-modules-core/commit/5544eace2be4c3e5192022a55c1dccdf70582c55).